mex/common/models/extracted_data.py

```diff
@@ -118,35 +121,71 @@ def set_identifiers(cls, values: dict[str, Any]) -> dict[str, Any]:
 
         # validate ID in primary source and primary source ID
         if identifier_in_primary_source := values.get("identifierInPrimarySource"):
-            identifier_in_primary_source = str(identifier_in_primary_source)
+            if isinstance(identifier_in_primary_source, list):
```

Comment:
The order of validators changed and now the input might be a list. I believe it was during the public-api tests. This should be fixed by using `@computed_field` because then pydantic will handle all the value cleaning for us.